test/unit/driver-specs.js

```diff
@@ -241,6 +243,28 @@ describe('XCUITestDriver', function () {
         );
         spy.notCalled.should.be.true;
       });
+
+      it('should throw an error if mjpegServerPort is occupied', async function () {
+        this.timeout(MOCHA_LONG_TIMEOUT);
+        delete device.simctl;
+        device.devicectl = true;
+        const net = require('net');
+        const server = net.createServer();
+        await new Promise((resolve) => server.listen(9100, resolve));
```

Comment:
we must also handle possible errors here, otherwise the promise will be stuck forever